Xamarin.PropertyEditing/ViewModels/PredefinedValuesViewModel.cs

```diff
@@ -16,42 +15,34 @@ public PredefinedValuesViewModel (TargetPlatform platform, IPropertyInfo propert
 			if (this.predefinedValues == null)
 				throw new ArgumentException (nameof(property) + " did not have predefined values", nameof(property));
 
+			var list = new List<string> (this.predefinedValues.PredefinedValues.Keys);
+			// If we're constrained but can't use the default, we need a blank to represent default
+			if (this.predefinedValues.IsConstrainedToPredefined) {
+				if (!property.ValueSources.HasFlag (ValueSources.Default) || !TryGetValueName (default(TValue), out string defaultName)) {
```

Comment:
Curious about why these aren't be combined with `&&`. Is it the pattern matching?